packages/ui/src/components/Card/Card.tsx

```diff
@@ -0,0 +1,60 @@
+import { Slot } from '@radix-ui/react-slot'
+import { type RecipeVariants } from '@vanilla-extract/recipes'
+import clsx from 'clsx'
+import { type ComponentProps, type PropsWithChildren } from 'react'
+import { Heading, Text } from 'ui'
+import { cardHeader, cardRoot } from './Card.css'
+
+type RootStyleProps = RecipeVariants<typeof cardRoot>
+type RootProps = ComponentProps<'div'> & RootStyleProps
+
+function CardRoot({ variant, size, className, children }: PropsWithChildren<RootProps>) {
+  return <article className={clsx(cardRoot({ variant, size }), className)}>{children}</article>
+}
+
+type HeaderProps = ComponentProps<'header'>
+
+function CardHeader({ className, children }: PropsWithChildren<HeaderProps>) {
+  return <header className={clsx(cardHeader, className)}>{children}</header>
+}
+
+type MediaProps = ComponentProps<typeof Slot>
+
+function CardMedia({ children }: PropsWithChildren<MediaProps>) {
+  return <Slot>{children}</Slot>
+}
+
+type HeadingProps = ComponentProps<'div'>
+
+function CardHeading({ className, children }: PropsWithChildren<HeadingProps>) {
+  return <div className={className}>{children}</div>
+}
+
+type TitleProps = ComponentProps<typeof Heading>
```

Comment:
What about composing the full type where it's declared?
```suggestion
type TitleProps = PropsWithChildren<ComponentProps<'div'>>
```

Reply:
Sure! good point, will update!